full-service/src/service/sync.rs

```diff
@@ -222,172 +222,93 @@ pub fn sync_account_next_chunk(
-            duration,
-            num_received_txos,
-            num_spent_txos,
-            unspent_key_images.len()
-        );
-            num_received_txos
+            (*view_account_key.view_private_key(), None)
         } else {
             let account_key: AccountKey = mc_util_serial::decode(&account.account_key)?;
+            (*account_key.view_private_key(), Some(account_key))
+        };
 
-            // Attempt to decode each transaction as received by this account.
-            let received_txos: Vec<_> = tx_outs
-                .into_par_iter()
-                .filter_map(|(block_index, tx_out)| {
-                    let amount = decode_amount(&tx_out, account_key.view_private_key())?;
-                    Some((block_index, tx_out, amount))
-                })
-                .collect();
-
-            let mut received_txos_with_subaddresses_and_key_images = Vec::new();
-            for (block_index, tx_out, amount) in received_txos {
-                let (subaddress_index, key_image) =
-                    decode_subaddress_and_key_image(&tx_out, &account_key, conn);
-
-                received_txos_with_subaddresses_and_key_images.push((
-                    block_index,
-                    tx_out,
-                    amount,
-                    subaddress_index,
-                    key_image,
-                ));
-            }
-
-            let num_received_txos = received_txos_with_subaddresses_and_key_images.len();
-
-            // Write received transactions to the database.
-            for (block_index, tx_out, amount, subaddress_index, key_image) in
-                received_txos_with_subaddresses_and_key_images
-            {
-                Txo::create_received(
-                    tx_out.clone(),
-                    subaddress_index,
-                    key_image,
-                    amount,
-                    block_index,
-                    account_id_hex,
-                    conn,
-                )?;
-            }
+        // Attempt to decode each transaction as received by this account.
+        let received_txos: Vec<_> = tx_outs
+            .into_par_iter()
+            .filter_map(|(block_index, tx_out)| {
+                let amount = decode_amount(&tx_out, &view_private_key)?;
+                Some((block_index, tx_out, amount))
+            })
+            .collect();
+
+        let mut received_txos_with_subaddresses_and_key_images = Vec::new();
+        for (block_index, tx_out, amount) in received_txos {
+            let (subaddress_index, key_image) = decode_subaddress_and_key_image(
+                &tx_out,
+                &view_private_key,
+                account_key.as_ref(),
+                conn,
+            );
+
+            received_txos_with_subaddresses_and_key_images.push((
+                block_index,
+                tx_out,
+                amount,
+                subaddress_index,
+                key_image,
+            ));
+        }
```

Comment:
it's unfortunate that github's diff logic started spreading out here as this is more or less the same
doing a diff locally in something like beyondcompare might be easier to visualize
![image](https://github.com/user-attachments/assets/cafffc4b-2ecd-4c1b-9b89-32192266ecdb)

Reply:
I can't get the above image to display as more than a thumbnail.  Am I missing something good? 😅

Reply:
sorry that's the sidebar from BeyondCompare, unfurtunatly the changes were so long that I couldn't get a nice screenshot of it